pkg/tsdb/influxdb/influxql/converter/converter_test.go

```diff
@@ -1,47 +1,100 @@
-package converter
+package converter_test
 
 import (
+	"io"
 	"os"
 	"path"
 	"strings"
 	"testing"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana-plugin-sdk-go/experimental"
-	jsoniter "github.com/json-iterator/go"
 	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/tsdb/influxdb/influxql"
+	"github.com/grafana/grafana/pkg/tsdb/influxdb/models"
 )
 
 const update = true
+const streamParser = false
+
+type testFile struct {
+	fileName string
+	rawQuery string
+}
 
-var files = []string{
-	"influx_select_all_from_cpu",
-	"influx_select_value_from_any",
+var files = []testFile{
+	{
+		fileName: "influx_select_all_from_cpu",
+		rawQuery: "select_all from cpu",
+	},
+	{
+		fileName: "select_value_from_cpu",
+		rawQuery: "select_value from cpu",
+	},
+	{
+		fileName: "select_multiple_from_cpu",
+		rawQuery: "select_multiple values from cpu",
+	},
+	{
+		fileName: "select_multiple_from_cpu_group_by_tag",
+		rawQuery: "select_multiple values from cpu grouped by a tag",
+	},
+	{
+		fileName: "select_value_from_measurement_with_labels",
+		rawQuery: "SELECT_value from measurement grouped by tag",
+	},
 }
 
 func TestReadPromFrames(t *testing.T) {
-	for _, name := range files {
-		t.Run(name, runScenario(name))
+	for _, f := range files {
+		t.Run(f.fileName, runScenario(f))
 	}
 }
 
 //lint:ignore U1000 Ignore used function for now
-func runScenario(name string) func(t *testing.T) {
+func runScenario(tf testFile) func(t *testing.T) {
 	return func(t *testing.T) {
 		// Safe to disable, this is a test.
 		// nolint:gosec
-		f, err := os.Open(path.Join("testdata", name+".json"))
+		f, err := os.Open(path.Join("testdata", tf.fileName+".json"))
 		require.NoError(t, err)
 
-		iter := jsoniter.Parse(jsoniter.ConfigDefault, f, 1024)
-		rsp := ReadInfluxQLStyleResult(iter)
+		shouldUpdate := update
+		var rsp backend.DataResponse
+
+		query := &models.Query{
+			Measurement:  "",
+			Policy:       "",
+			Tags:         nil,
+			GroupBy:      nil,
+			Selects:      nil,
+			RawQuery:     tf.rawQuery,
+			UseRawQuery:  true,
+			Alias:        "",
+			Interval:     0,
+			Tz:           "",
+			Limit:        "",
+			Slimit:       "",
+			OrderByTime:  "",
+			RefID:        "",
+			ResultFormat: "time_series",
+		}
+
+		if streamParser {
+			// shouldUpdate = false
+			rsp = influxql.StreamParse(f, 200, query)
+		} else {
+			rsp = *influxql.ResponseParse(io.NopCloser(f), 200, query)
```

Comment:
So current response parser returns a pointer and the new parser doesn't. I am not sure which one is better. (A bit confused too). I am open to suggestions.